### ISSUE TITLE
comment_count doesn't get updated when deleting a discussion

### ISSUE THREAD

**Issue by iPurpl3x** [flarum/core#2227]
## Bug Report

**Current Behavior**
When deleting the discussion, the comment count (`users.comment_count` in the database) doesn't get updated. Only after creating a new discussion, this will get a correct value again.

**Steps to Reproduce**
1. Log into an empty instance, as Admin
2. Create a new discussion
3. Check the profile page : should have **Posts 1** and **Discussions 1**.
4. Delete the discussion that was created (delete forever)
5. Check the profile page : should have **Posts 1** and **Discussions 0**.

**Expected Behavior**
At step 5. there should be  **Posts 0**

**Environment**
- Flarum version: 0.1.0-beta.13
- Website URL: localhost
- Webserver: nginx
- Hosting environment: local
- PHP version: 7.3.19
- Browser: Google Chrome Version 83.0.4103.116

_(I Hope this isn't a duplicate, I didn't find anything when searching for this issue...)_

**Comment by clarkwinkelmann** [comment]
I can replicate the issue.

Looking at the metadata updater:

https://github.com/flarum/core/blob/d492579638fb52dafbfe65f1f36a95eb6047f7f3/src/User/UserMetadataUpdater.php#L52-L63

Both `whenDiscussionWasStarted` and `whenDiscussionWasDeleted` trigger an update of the discussion count but not of the comment count.

It's not a problem in `whenDiscussionWasStarted` because a separate event is triggered when the post is created and we already react to that event a few lines above in the same file.

But when a discussion delete happens, no delete event is triggered on the posts which is why nothing happens.

I think we can simply solve this by calling `updateCommentsCount` in `whenDiscussionWasDeleted`, changing the signature of the method to accept both a Post or a Discussion.

---

Digging into this I realized to my surprise that the comment_count is the total number of posts of type `comment` that aren't private. There is no regard to whether the posts are hidden, or whether they belong to a hidden discussion. But this doesn't have anything to do with the issue at hand.

**Comment by iPurpl3x** [comment]
I noticed that thing about the hidden posts too @clarkwinkelmann, it's kind of confusing for the user. Should we open another issue for that?

**Comment by clarkwinkelmann** [comment]
@iPurpl3x if you can do that, yes please :+1:

**Comment by clarkwinkelmann** [comment]
Still relevant, also doesn't seem too complicated to solve.
